src/KafkaFactory.ts

```diff
@@ -0,0 +1,33 @@
+import kafkaDojot = require("@dojot/adminkafka");
+import { logger } from "@dojot/dojot-module-logger";
+import kafka = require("kafka-node");
+import config = require("./config");
+
+export class KafkaFactory {
+    private host?: string;
+
+    constructor(theHost?: string) {
+        this.host = theHost;
+    }
+
+    public client(): kafka.KafkaClient {
+        logger.debug("Creating new Kafka producer...", {filename: "producer"});
+        const kafkaHost = this.host ? this.host : config.kafka.kafkaAddress + ":" + config.kafka.kafkaPort.toString();
+        logger.debug("Creating Kafka client...", {filename: "producer"});
+        return new kafka.KafkaClient({kafkaHost});
+    }
+
+    public dojotProducer(): kafkaDojot.Admin {
+        logger.debug("... Kafka client was created.", {filename: "producer"});
```

Comment:
Isn't this line misplaced? It feels like it should be in client() method.

Reply:
Thank you @giovannicuriel. I fixed it.